homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -211,13 +211,16 @@ async def _async_update_others(self) -> None:
             log_update_error(_LOGGER, "update", self.name, "binary sensor", error)
             return
 
-        if (event_code := self.entity_description.event_code) is None:
+        if (event_codes := self.entity_description.event_codes) is None or len(
+            event_codes
+        ) == 0:
             _LOGGER.error("Binary sensor %s event code not set", self.name)
```

Comment:
Side note: This would be a developer error and then we should raise an exception, not log an error.

Reply:
That sounds good, I changed this over to raise a ValueError.